diff -Naur embree-3.5.2_SRC_orig/common/sys/alloc.cpp embree-3.5.2_SRC_modif/common/sys/alloc.cpp
--- embree-3.5.2_SRC_orig/common/sys/alloc.cpp	2019-03-13 14:13:56.000000000 +0300
+++ embree-3.5.2_SRC_modif/common/sys/alloc.cpp	2019-07-23 12:59:36.127707285 +0300
@@ -22,6 +22,10 @@
 ////////////////////////////////////////////////////////////////////////////////
 /// All Platforms
 ////////////////////////////////////////////////////////////////////////////////
+
+#ifndef MAP_HUGETLB
+#define MAP_HUGETLB     0x40000         /* create a huge page mapping */
+#endif
   
 namespace embree
 {
