--- netgen-6.2.2101_orig/CMakeLists.txt	2021-09-21 17:05:13.000000000 +0300
+++ netgen-6.2.2101_new/CMakeLists.txt	2021-10-25 16:31:45.656379721 +0300
@@ -233,6 +233,11 @@
 endif(NOT ZLIB_INCLUDE_DIRS)
 include_directories(${ZLIB_INCLUDE_DIRS})
 
+find_package(Threads REQUIRED)
+if(NOT WIN32)
+  string(APPEND CMAKE_CXX_STANDARD_LIBRARIES " -dl")
+endif()
+
 #######################################################################
 if (USE_GUI)
     find_package(TCL 8.5 REQUIRED)
