diff -Naur opencv-3.2.0_SRC_orig/cmake/OpenCVCompilerOptions.cmake opencv-3.2.0_SRC_modif/cmake/OpenCVCompilerOptions.cmake
--- opencv-3.2.0_SRC_orig/cmake/OpenCVCompilerOptions.cmake	2019-09-02 17:10:46.940341420 +0300
+++ opencv-3.2.0_SRC_modif/cmake/OpenCVCompilerOptions.cmake	2019-09-02 17:12:49.592741875 +0300
@@ -18,9 +18,9 @@
         message(STATUS "Unable to compile program with enabled ccache, reverting...")
         set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${__OLD_RULE_LAUNCH_COMPILE}")
       endif()
-    else()
-      message(STATUS "Looking for ccache - not found")
     endif()
+  else()
+    message(STATUS "Looking for ccache - not found")
   endif()
 endif()
 
