diff -Naur opencv-3.2.0_SRC_orig/cmake/OpenCVCompilerOptions.cmake opencv-3.2.0_SRC_modif/cmake/OpenCVCompilerOptions.cmake
--- opencv-3.2.0_SRC_orig/cmake/OpenCVCompilerOptions.cmake	2016-12-23 15:54:44.000000000 +0300
+++ opencv-3.2.0_SRC_modif/cmake/OpenCVCompilerOptions.cmake	2019-11-14 17:51:34.265254970 +0300
@@ -28,17 +28,17 @@
   set(ENABLE_PRECOMPILED_HEADERS OFF CACHE BOOL "" FORCE)
 endif()
 
-if(MINGW OR (X86 AND UNIX AND NOT APPLE))
-  # mingw compiler is known to produce unstable SSE code with -O3 hence we are trying to use -O2 instead
+if(1 OR MINGW OR (X86 AND UNIX AND NOT APPLE))
+  # mingw compiler is known to produce unstable SSE code with -O3 hence we are trying to use -O0 instead
   if(CMAKE_COMPILER_IS_GNUCXX)
     foreach(flags CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_RELEASE CMAKE_CXX_FLAGS_DEBUG)
-      string(REPLACE "-O3" "-O2" ${flags} "${${flags}}")
+      string(REPLACE "-O3" "-O0" ${flags} "${${flags}}")
     endforeach()
   endif()
 
   if(CMAKE_COMPILER_IS_GNUCC)
     foreach(flags CMAKE_C_FLAGS CMAKE_C_FLAGS_RELEASE CMAKE_C_FLAGS_DEBUG)
-      string(REPLACE "-O3" "-O2" ${flags} "${${flags}}")
+      string(REPLACE "-O3" "-O0" ${flags} "${${flags}}")
     endforeach()
   endif()
 endif()
