--- ParaView_origin/VTK/Rendering/VolumeOpenGL2/vtkOpenGLGPUVolumeRayCastMapper.cxx	2020-06-18 14:59:43.630941741 +0200
+++ ParaView/VTK/Rendering/VolumeOpenGL2/vtkOpenGLGPUVolumeRayCastMapper.cxx	2020-06-18 15:00:57.582622827 +0200
@@ -2822,7 +2822,7 @@
     this->VolumePropertyChanged |= property->GetMTime() > this->ShaderBuildTime.GetMTime();
 
     auto it = this->Parent->AssembledInputs.find(port);
-    if (it == this->Parent->AssembledInputs.cend())
+    if (it == this->Parent->AssembledInputs.cend() || it->second.Volume != vol)
     {
       // Create new input structure
       auto texture = vtkSmartPointer<vtkVolumeTexture>::New();
