diff -ruN ParaView/VTK/Utilities/PythonInterpreter/vtkPythonInterpreter.cxx ParaView_new/VTK/Utilities/PythonInterpreter/vtkPythonInterpreter.cxx
--- ParaView/VTK/Utilities/PythonInterpreter/vtkPythonInterpreter.cxx	2019-05-28 08:19:55.673481437 +0200
+++ ParaView_new/VTK/Utilities/PythonInterpreter/vtkPythonInterpreter.cxx	2019-05-28 08:08:21.000000000 +0200
@@ -260,8 +260,11 @@
     vtkPythonInterpreter::InitializedOnce = true;
 
 #ifdef VTK_PYTHON_FULL_THREADSAFE
-    PyEval_InitThreads(); // safe to call this multiple time
-    PyEval_SaveThread(); // release GIL
+    if(PyEval_ThreadsInitialized() == 0)
+    {
+      PyEval_InitThreads(); // safe to call this multiple time
+      PyEval_SaveThread(); // release GIL
+    }
 #endif
 
     // HACK: Calling PyRun_SimpleString for the first time for some reason results in
