diff -Naur ParaView-5.6.0-bfaf7b82_SRC_orig/VTK/Rendering/OpenGL2/vtkStateStorage.h ParaView-5.6.0-bfaf7b82_SRC_modif/VTK/Rendering/OpenGL2/vtkStateStorage.h
--- ParaView-5.6.0-bfaf7b82_SRC_orig/VTK/Rendering/OpenGL2/vtkStateStorage.h	2019-06-25 16:00:38.000000000 +0300
+++ ParaView-5.6.0-bfaf7b82_SRC_modif/VTK/Rendering/OpenGL2/vtkStateStorage.h	2019-07-12 11:32:00.000000000 +0300
@@ -51,8 +51,11 @@
 
 #include <algorithm>
 #include <string>
+#include <vector>
 
-#ifndef NDEBUG // a debug implementation
+// uncomment the folowing line to add in state debugging information
+//#define USE_STATE_DEBUGGING 1
+#ifdef USE_STATE_DEBUGGING
 
 class VTKRENDERINGOPENGL2_EXPORT vtkStateStorage
 {
@@ -120,7 +123,7 @@
   this->Storage.insert(this->Storage.end(), start, start + sizeof(T));
 }
 
-#else // release implementation
+#else // normal implementation
 
 class VTKRENDERINGOPENGL2_EXPORT vtkStateStorage
 {
@@ -157,7 +160,7 @@
   this->Storage.insert(this->Storage.end(), start, start + sizeof(T));
 }
 
-#endif // Release implementation
+#endif // normal implementation
 
 #endif // vtkStateStorage_h
 
