--- PyQt5_sip-12.8.1/sipint.h	2020-08-29 13:27:37.000000000 +0300
+++ PyQt5_sip-12.8.1/sipint.h.new	2021-03-17 12:06:11.287544763 +0300
@@ -177,9 +177,9 @@
 void sipOMAddObject(sipObjectMap *om, sipSimpleWrapper *val);
 int sipOMRemoveObject(sipObjectMap *om, sipSimpleWrapper *val);
 
-#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+//#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 #define sipSetBool(p, v)    (*(_Bool *)(p) = (v))
-#endif
+//#endif
 
 
 #ifdef __cplusplus
