--- rkcommon-1.5.1/cmake/FindTBB.cmake	2021-03-09 14:26:46.000000000 +0300
+++ rkcommon-1.5.1/cmake/FindTBB.cmake.new	2021-03-17 12:26:35.339966172 +0300
@@ -269,8 +269,7 @@
 function(rk_tbb_find_include_directory)
   find_path(TBB_INCLUDE_DIR
     NAMES "${_TBB_HEADER}"
-    HINTS "${TBB_ROOT}/${_TBB_INCLUDE_SUBDIR}"
-    NO_PACKAGE_ROOT_PATH)
+    HINTS "${TBB_ROOT}/${_TBB_INCLUDE_SUBDIR}")
 endfunction()
 
 #===============================================================================
