diff -Naur tbb-2019-U8_SRC_orig/build/build.py tbb-2019-U8_SRC_modif/build/build.py
--- tbb-2019-U8_SRC_orig/build/build.py	2019-06-06 13:29:15.000000000 +0300
+++ tbb-2019-U8_SRC_modif/build/build.py	2019-11-15 14:54:44.242189708 +0300
@@ -198,7 +198,7 @@
     paths = [os.path.abspath(d) for d in [args.prefix, inc_dir, irml_dir, lib_dir]+release_dirs]
     os.environ["TBBROOT"] = paths[0]
     # all the paths must be relative to python/ directory or be absolute
-    assert system('python python/setup.py build -b%s build_ext -I%s -L%s install -f'% \
-        (paths[2], paths[1], ':'.join(paths[2:]))) == 0
+    assert system('python3 python/setup.py build -b%s build_ext -I%s -L%s install -f --prefix=%s'% \
+        (paths[2], paths[1], ':'.join(paths[2:]), args.prefix)) == 0
 
 print("done")
