commit 43401eee36e1af2ca170b2824e76875f2ac56964
Author: spo <sergey.pokhodenko@opencascade.com>
Date:   Tue Dec 22 12:46:02 2015 +0300

    Replace nullptr with 0

diff --git a/Eigen/src/Core/DenseStorage.h b/Eigen/src/Core/DenseStorage.h
index 568493c..c30c084 100755
--- a/Eigen/src/Core/DenseStorage.h
+++ b/Eigen/src/Core/DenseStorage.h
@@ -278,7 +278,7 @@ template<typename T, int _Options> class DenseStorage<T, Dynamic, Dynamic, Dynam
       , m_rows(std::move(other.m_rows))
       , m_cols(std::move(other.m_cols))
     {
-      other.m_data = nullptr;
+      other.m_data = 0;
     }
     DenseStorage& operator=(DenseStorage&& other)
     {
@@ -336,7 +336,7 @@ template<typename T, int _Rows, int _Options> class DenseStorage<T, Dynamic, _Ro
       : m_data(std::move(other.m_data))
       , m_cols(std::move(other.m_cols))
     {
-      other.m_data = nullptr;
+      other.m_data = 0;
     }
     DenseStorage& operator=(DenseStorage&& other)
     {
@@ -390,7 +390,7 @@ template<typename T, int _Cols, int _Options> class DenseStorage<T, Dynamic, Dyn
       : m_data(std::move(other.m_data))
       , m_rows(std::move(other.m_rows))
     {
-      other.m_data = nullptr;
+      other.m_data = 0;
     }
     DenseStorage& operator=(DenseStorage&& other)
     {
