--- A/CMakeLists.txt	2016-07-19 11:43:42.845939766 +0200
+++ B/CMakeLists.txt	2016-07-19 11:44:33.633939452 +0200
@@ -140,9 +140,9 @@
 
 if(GLUT_FOUND)
   add_executable(gl2psTest WIN32 gl2psTest.c)
-  target_link_libraries(gl2psTest lib ${EXTERNAL_LIBRARIES})
+  target_link_libraries(gl2psTest lib m ${EXTERNAL_LIBRARIES})
   add_executable(gl2psTestSimple WIN32 gl2psTestSimple.c)
-  target_link_libraries(gl2psTestSimple lib ${EXTERNAL_LIBRARIES})
+  target_link_libraries(gl2psTestSimple lib m ${EXTERNAL_LIBRARIES})
 endif(GLUT_FOUND)
 
 find_package(LATEX)
