--- a/setup.py	2009-11-15 17:06:10.000000000 +0100
+++ b/setup.py	2012-09-19 13:39:39.417889941 +0200
@@ -34,8 +34,13 @@
 # TIFF_ROOT = libinclude("/opt/tiff")
 
 TCL_ROOT = None
+TCL_ROOT = libinclude(os.getenv("TCLDIR"))
+TK_ROOT = libinclude(os.getenv("TKDIR"))
 JPEG_ROOT = None
 ZLIB_ROOT = None
+JPEG_ROOT = ("/usr/lib/x86_64-linux-gnu","/usr/include/x86_64-linux-gnu")
+ZLIB_ROOT = ("/usr/lib/x86_64-linux-gnu","/usr/include/x86_64-linux-gnu")
 TIFF_ROOT = None
 FREETYPE_ROOT = None
+FREETYPE_ROOT = libinclude(os.getenv("FREETYPEDIR"))
 LCMS_ROOT = None
@@ -189,7 +193,7 @@
         #
         # add configured kits
 
-        for root in (TCL_ROOT, JPEG_ROOT, TCL_ROOT, TIFF_ROOT, ZLIB_ROOT,
+        for root in (TCL_ROOT, JPEG_ROOT, TK_ROOT, TIFF_ROOT, ZLIB_ROOT,
                      FREETYPE_ROOT, LCMS_ROOT):
             if isinstance(root, type(())):
                 lib_root, include_root = root
