--- a/setup.py	2016-09-29 14:48:18.045084416 +0200
+++ b/setup.py	2016-09-29 14:49:30.645413214 +0200
@@ -113,7 +113,8 @@
 
     r_libs = [ # Different verisons of R put .so/.dll in different places
               os.path.join(RHOME, 'bin'),  # R 2.0.0+
-              os.path.join(RHOME, 'lib'),  # Pre 2.0.0
+              os.path.join(RHOME, 'lib64', 'R', 'lib'),  # Pre 2.0.0
+              os.path.join(RHOME, 'lib', 'R', 'lib'),  # Pre 2.0.0
              ]
 
     print "RHOME=",RHOME
@@ -158,8 +159,9 @@
         extra_compile_args=["-shared"]
         source_files = source_files + ["src/setenv.c"]
     else: # unix-like systems, this is known to work for Linux and Solaris
-        include_dirs = [ os.path.join(RHOME.strip(), 'include'), 
-                         'src', '/usr/share/R/include' ]
+        include_dirs = [ os.path.join(RHOME.strip(), 'lib64' ,'R' ,'include'), 
+                         os.path.join(RHOME.strip(), 'lib' ,'R' ,'include'), 
+                         'src' ]
         libraries=['R','Rlapack']
         library_dirs = r_libs
         runtime_libs = r_libs
